irods_specific_implementation/irods_rest_meta_api_implementation.h

```diff
@@ -0,0 +1,161 @@
+#include <irods/metadata.hpp>
+
+#include <pistache/router.h>
+#include <nlohmann/json.hpp>
+
+
+#define OP_STATUS_SUCCESS "SUCCESS"
+#define OP_STATUS_FAILED "FAILURE"
+
+namespace irods::rest
+{
+
+    namespace ixm = irods::experimental::metadata;
+
+    // this is contractually tied directly to the api implementation
+    const std::string service_name{"irods_rest_cpp_meta_server"};
+
+    class meta : public api_base
+    {
+    public:
+        meta()
+            : api_base{service_name}
+        {
+            info("Endpoint initialized.");
+        }
+
+        std::tuple<Pistache::Http::Code, std::string>
+        operator()(const Pistache::Rest::Request& _request,
+                   Pistache::Http::ResponseWriter& _response)
+        {
+            try {
+                auto _attrib       = _request.query().get("attrib").get();
+                auto _value        = _request.query().get("value").get();
+                auto _unit         = _request.query().get("unit").get();
+                auto _entity_type  = _request.query().get("entity_type").get();
+                auto _sub_cmd      = _request.query().get("sub_cmd").get();
+                auto _target       = _request.query().get("target").get();
+
+                // unsure what the hint parameter achieves here
+                auto conn = get_connection(_request.headers().getRaw("authorization").value());
+
+                nlohmann::json json_obj;
+
+                // construct avu struct
+                ixm::avu triple {
+                    _attrib,
+                    _value,
+                    _unit
+                };
+                // i think the target parameter is constructed at a lower level of abstraction
+                // get entity type
+                auto et = ixm::to_entity_type("-" + _entity_type);
+
+                if (_sub_cmd == "add"){
+                        ixm::add(
+                            *conn(),
+                            triple,
+                            et,
+                            _target
+                        );
+                        json_obj["op"] = "add";
+                        json_obj["op_status"] = OP_STATUS_SUCCESS;
+                        json_obj["avu"] = {
+                            { "attrib", _attrib },
+                            { "value", _value },
+                            { "unit", _unit }
+                        };
+                        json_obj["target"] = _target;
+                        json_obj["entity_type"] = _entity_type;
+                        return std::make_tuple(
+                            Pistache::Http::Code::Ok,
+                            json_obj.dump(
+                                -1,
+                                ' ',
+                                false,
+                                nlohmann::json::error_handler_t::ignore
+                            )
+                        );
+                } else if (_sub_cmd == "rm") {
```

Comment:
Swap order of operands used in the comparison.

*Applies to other if-statements too*

Reply:
Happy to do so. What's the benefit to this?

Reply:
If `_sub_cmd` is not `const` and `=` is used (accidentally), you end up with a silent bug. Swapping the order of these operands guards against that issue because `"rm" cannot be assigned to and leads to a compiler error.